core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -1029,14 +1045,124 @@ struct ParallelReduceSpecialize<FunctorType, TeamPolicyInternal<PolicyArgs...>,
     void* scratch_ptr = OpenMPTargetExec::get_scratch_ptr();
 
     ValueType result = ValueType();
-#pragma omp target teams distribute parallel for num_teams(nteams) num_threads(team_size*vector_length) \
-         map(to:f,scratch_ptr) map(tofrom:result) reduction(+: result) schedule(static,1)
-    for (int i = 0; i < league_size * team_size * vector_length; i++) {
-      typename PolicyType::member_type team(i / (team_size * vector_length),
+
+#pragma omp target teams distribute num_teams(nteams) thread_limit(team_size) \
+         map(to:f,scratch_ptr) map(tofrom:result) reduction(+: result)
+    for (int i = 0; i < league_size ; i++) {
+      ValueType inner_result = ValueType();
+#pragma omp parallel \
+    reduction(+:inner_result)
+      {
+      typename PolicyType::member_type team(i ,
                                             league_size, team_size,
                                             vector_length, scratch_ptr, 0, 0);
-      f(team, result);
-      if (team.m_vector_lane != 0) result = 0;
+      f(team, inner_result);
+//      if (team.m_vector_lane != 0) result = 0;
```

Comment:
Why did you comment that line?

Reply:
Yes this was required in the earlier implementation where we launched team_size*vector_size number of threads per team and did not do anything special for ThreadVector parallelism. I will delete that line.